source/uri/Endpoint.ooc

```diff
@@ -0,0 +1,47 @@
+Endpoint: class {
+	_host: VectorList<Text>
+	_port: Int
+	host: VectorList<Text> {
+		get {
+			result := VectorList<Text> new()
+			for (i in 0 .. this _host count)
+				result add(this _host[i] take())
+			result
+		}
+	}
+	port ::= this _port
+
+	init: func (=_host, =_port)
+	free: override func {
+		for (i in 0 .. this _host count)
+			this _host[i] free(Owner Receiver)
+		this _host free()
+		super()
+	}
+	toText: func -> Text {
+		result := Text new()
+		for (i in 0 .. this _host count)
+			result += i == 0 ? this _host[i] take() : t"." + this _host[i] take()
```

Comment:
`this _host` will probably not be very large (maybe around 3 items) and this code will probably only run once or a few times when used. I don't think much is gained by using it here or in other places in this PR.